config/src/main/java/org/springframework/security/config/annotation/web/configurers/X509Configurer.java

```diff
@@ -177,8 +179,16 @@ public X509Configurer<H> subjectPrincipalRegex(String subjectPrincipalRegex) {
 	public void init(H http) {
 		PreAuthenticatedAuthenticationProvider authenticationProvider = new PreAuthenticatedAuthenticationProvider();
 		authenticationProvider.setPreAuthenticatedUserDetailsService(getAuthenticationUserDetailsService(http));
+		authenticationProvider.setGrantedAuthoritySupplier(() -> AuthorityUtils.createAuthorityList("FACTOR_X509"));
 		http.authenticationProvider(authenticationProvider)
 			.setSharedObject(AuthenticationEntryPoint.class, new Http403ForbiddenEntryPoint());
+		ExceptionHandlingConfigurer<H> exceptions = http.getConfigurer(ExceptionHandlingConfigurer.class);
+		if (exceptions != null) {
+			AuthenticationEntryPoint forbidden = new Http403ForbiddenEntryPoint();
+			exceptions.defaultAuthenticationEntryPointFor(forbidden, AnyRequestMatcher.INSTANCE);
+			exceptions.defaultAuthenticationEntryPointFor(
+					(ep) -> ep.addEntryPointFor(forbidden, AnyRequestMatcher.INSTANCE), "FACTOR_X509");
```

Comment:
```suggestion
			exceptions.defaultAuthenticationEntryPointFor(
					(ep) -> ep.defaultEntryPoint(forbidden), "FACTOR_X509");
```

This seems a bit more concise and does the same thing.

Reply:
As far as I can tell, this causes the `DelegatingAuthenticationEntryPoint` with throw an exception since `List<RequestMatcherEntry>` is empty. That's because `.addEntryPointFor` is never called for this builder elsewhere in the code.